influxdata/encoder.go

```diff
@@ -125,8 +125,8 @@ func (e *Encoder) StartLine(measurement string) {
 			return
 		}
 	}
-	if section != MeasurementSection {
-		// This isn't the first line, so add a newline separator.
+	if section != MeasurementSection && len(e.buf) > 0 && e.buf[len(e.buf)-1] != '\n' {
```

Comment:
I think we can be a tiny bit nicer here, because we know that newline has been added iff the section is currently `endSection`.